components/micro-gateway-core/src/main/ballerina/gateway/oauthn_handler.bal

```diff
@@ -258,6 +162,53 @@ public function OAuthAuthProvider::authenticate (APIRequestMetaDataDto apiReques
     return apiKeyValidationDto;
 }
 
+public function OAuthAuthProvider::invokeKeyValidation(APIRequestMetaDataDto apiRequestMetaDataDto) returns (boolean,
```

Comment:
Add doc comment since this is a public method.

Reply:
I will add this while incorpertating https://github.com/wso2/product-microgateway/pull/144 with changes in this PR

Reply:
My bad. Forgot to add it in 144. Will add it in my next PR.

Reply:
Added with https://github.com/wso2/product-microgateway/pull/158

Reply:
Thanks @malinthaprasan for refactoring based on comments